### DISCUSSION TITLE
Importing bootstrap overriding default CSS

### DISCUSSION THREAD

**Discussion by Laharikota07** [tinymce/tinymce#8719] [General]
Table borders, cell borders, and other default styles of Tiny MCE are overridden when importing bootstrap using the content_css attribute.

 Steps to reproduce issue:
    1. Go to tiny MCE editor and  add table.
    2. Then try to add border style to the table using table properties and style wont get applied for display.
 

BootStrap CSS Import
![image](https://user-images.githubusercontent.com/84136998/236801518-f2041b72-6315-4a7d-8b06-81cd61f9054c.png)

 Original Table
<img width="857" alt="image" src="https://user-images.githubusercontent.com/84136998/237004982-e90535f0-4146-49fd-a162-6b7dcc2d4355.png">

 
Source Code View after adding the Border Style & Border Color 
![image](https://user-images.githubusercontent.com/84136998/237005224-b75dff19-95f7-4524-8974-e7c011332c0c.png) 

Expected behavior:
    1. Display in Editor as well as Preview should take into account the Styles set in Editor via table properties.
   

Actual Behaviour:
 1. The Styles are not taking affect. NO effect of Dashed mode is visible.
![image](https://user-images.githubusercontent.com/84136998/237007017-378d525c-c8d8-4e4b-be6b-b2eb19a8a00a.png)

Versions using:
    1. Angular Packages  : 15.X
    3. "@ng-bootstrap/ng-bootstrap": "^13.1.1"
    4. bootsrap: ^5.1.3   
    5. "tinymce": "^6.3.2"
    6. "@tinymce/tinymce-angular": "^7.0.0"

**Comment by lostkeys** [comment]
Hi @Laharikota07. 

The issue you face is an effect of how tables are styled using CSS in general and not having anything with bootstrap (since you don't have the table class on the table element usually required by Bootstrap). The color is applied to cells as well, which negates the border style set on the table element. You can play around with this yourself. The solution is to make your own `content_css` that has only the table styles you need in the configuration that suits your use-case. 

To use Bootstrap with TinyMCE, you can [explore the ideas in this example](https://fiddle.tiny.cloud/r6iaab), on how to force classes on elements.

— Fredrik, Product Manager

**Reply by Laharikota07** [reply]
Also, when we give some border style to a particular cell of a table, the border style isn't getting applied to all 4 borders of that cell. Verified with the example editor you have shared. But it is not working.
Explain this too. And also, what could be done for that.

Thanks.

**Reply by lostkeys** [reply]
This is unfortunately how HTML and CSS works, and as I wrote below, any solution to this would require us to make a lot of decisions on how tables can and cannot be constructed, and getting it right is not an easy task.

**Comment by parulshrivastava455** [comment]
Hi Fredrik,

Given example by you is also not working. I am trying to add border style to the table by using Advance property, but any border is not getting applied. 
Waiting for your reply. 
Thanks in advance.

**Comment by lostkeys** [comment]
Hi @parulshrivastava455. Table styling has a 1:1 mapping with the underlying inline styles. So if you add a border color to a table tow the styles are applied to the `<tr>`. This means the user needs some understanding to properly style tables. There are also browser differences. Can you clarify more in detail exactly what steps you are performing?

**Reply by parulshrivastava455** [reply]
I have selected table.
Clicked on table ->table properties -> Advanced

<img width="914" alt="image" src="https://github.com/tinymce/tinymce/assets/136440216/adeb011f-88bb-4aca-99d4-0b64cc58412b">

using advance property, trying to add border style but border style is not appearing in the editor.

**Reply by lostkeys** [reply]
@parulshrivastava455 The default table styles apply borders on the `<td>` element which take precedence of any styles added on the `<table>` element. Either use the table cell properties dialog or modify the table styles in the content_css to fit your use-case.

**Comment by lostkeys** [comment]
I just want to add that we recognize styling tables are not very user friendly but any solution inevitable requires us to make a very opinionated implementation which many developers do not like either.
